docs/adr/adr-0023-adding-stylelint.md

```diff
@@ -0,0 +1,234 @@
+    "media-feature-range-operator-space-before": "always",
+    "media-feature-range-operator-space-after": "always",
+    "media-feature-parentheses-space-inside": "never",
+    "media-feature-name-no-vendor-prefix": true,
+    "media-feature-colon-space-before": "never",
+    "media-feature-colon-space-after": "never",
+    "no-descending-specificity": null,
+    "no-empty-first-line": null,
+    "declaration-block-no-redundant-longhand-properties": null,
+    "no-missing-end-of-source-newline": null,
+    "no-eol-whitespace": null,
+    "property-no-vendor-prefix": null,
+    "selector-class-pattern": [
+      "^([a-z][a-z0-9]*)((-{1,2}|_{1,2})[a-z0-9]+)*$",
+      {
+        "message": "Expected class selector to be either kebab-case, kebab--case or snake__case"
+      }
+    ],
+    "selector-id-pattern": [
+      "^([a-z][a-z0-9]*)((-{1,2}|_{1,2})[a-z0-9]+)*$",
+      {
+        "message": "Expected id selector to be either kebab-case, kebab--case or snake__case"
+      }
+    ]
+  }
+}
+```
+
+While we won't dive into the configuration deeper you can find the rules on the Stylelint website. 
+What we have done here is adding or overriding existing rules on the `Standard` configuration to best match our needs.
+
+To run and check our styles on our component we created two commands in our `package.json` file who are : 
+* `stylelint` which runs Stylelint and checks for errors 
+* `stylelint:fix` which runs Stylelint and checks for errors and auto-fixes the ones that are auto fixable
+* `stylelint:ci` which runs Stylelint and checks for errors and formats the output for our GitHub actions CI.
+
+The benefits of Stylelint are that we now have a coherent style across our component library code base and use the same format everywhere.
+It also allowed us to get rid of the custom ESLint sort plugin which is done by Stylelint directly from now on. 
+
+## Problems encountered
+
+While we are satisfied with our current configuration we encountered some problems in the process with some rules.
+
+### The whitespace rule
+
+First, we had some problems with the whitespace rule (no-eol-whitespace-rule). The purpose of this rule is not to have whitespace after a line ending (e.g: semicolon or empty lines). The problem with this is that the processor we use targets only the CSS inside template literals which means the backtick wasn't included.
+This leads the code going from this: 
+```javascript
+static get styles () {
+    return [
+      skeletonStyles,
+      // language=CSS
+      css`
+        :host {
+          display: block;
+        }
+      `,
+    ];
+```
+to this:
+```javascript
+static get styles () {
+    return [
+      skeletonStyles,
+      // language=CSS
+      css`
+        :host {
+          display: block;
+        }
+`,
+    ];
+```
+As you can see the ending backtick of the template literal isn't formatted correctly because of this rule as the rule is triggered as an empty line with whitespaces in it.
```

Comment:
```suggestion
As you can see, the closing backtick of the template literal isn't formatted correctly.
This is because Stylelint considers it as an empty line containing whitespaces.
```
(I tried to change the sentence but I'm not sure I understand it :sweat_smile: If my attempt is bad and you are stuck, we can discuss it and come up with something together :wink:)